### PR TITLE
[16.0][MIG] hr_phone: Migration to 16.0

### PR DESCRIPTION
**Pull request by nguyenminhchien**
This PR replaces https://github.com/OCA/connector-telephony/pull/295
Depend:
* https://github.com/OCA/connector-telephony/pull/296

### CODE REVIEW

**Comment by StephaneMangin** [comment]
For the PR dependency, please take in account this comment https://github.com/OCA/edi/pull/724#issuecomment-1461765822

**Comment by mmequignon** [comment]
<s>Could you please address this comment?
https://github.com/OCA/connector-telephony/pull/296#issuecomment-1543517072</s>

**Comment by nilshamerlinck** [comment]
Hello @mmequignon 
- your remark in previous PR has been taken into account: the old mixin has been removed in this PR and the code relies on odoo native's [`_phone_format`](https://github.com/odoo/odoo/blob/16.0/addons/phone_validation/models/res_partner.py#L21) that uses `odoo.addons.phone_validation.tools.phone_format` indeed
- `base_phone` comes with some other features (wizards, widget), and is needed for other modules such `crm_phone`
- that said if you think it's worth it, we will work on a new PR for an `hr_phone` that doesn't depend on `base_phone` anymore

**Comment by mmequignon** [comment]
> Hello @mmequignon
> 
>     * your remark in previous PR has been taken into account: the old mixin has been removed in this PR and the code relies on odoo native's [`_phone_format`](https://github.com/odoo/odoo/blob/16.0/addons/phone_validation/models/res_partner.py#L21) that uses `odoo.addons.phone_validation.tools.phone_format` indeed
> 
>     * `base_phone` comes with some other features (wizards, widget), and is needed for other modules such `crm_phone`
> 
>     * that said if you think it's worth it, we will work on a new PR for an `hr_phone` that doesn't depend on `base_phone` anymore

my bad, I thought the previous pr has just been reopened.
I'll update my review, please ignore my previous comment.
Sorry for the noise!

**Review by mmequignon — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+odoo-addon-base_phone @git+https://github.com/OCA/connector-telephony@refs/pull/296/head#subdirectory=setup/base_phone
```
Comment (florentx) [inline test-requirements.txt]:
```suggestion
odoo-addon-base_phone @git+https://github.com/OCA/connector-telephony@refs/pull/300/head#subdirectory=setup/base_phone
```

**Review by florentx — APPROVED** [review]
Please adapt:

- [ ] OCA/connector-telephony#300

File: test-requirements.txt (hunk 1 above)
Comment (nguyenminhchien) [reply test-requirements.txt]: @florentx : updated

**Pull request closed without merging**